API/events/eventManager.m

```diff
@@ -36,62 +36,56 @@ function setEvents(value)
             value = events;
         end
         
-        function value = getHandlers()
-            % Gets the list of registered callbacks
+        function funcID = getCallbackID(callback)
+            % Generates a id for a callback handle function
             %
-            % handlerList = eventManager.getHandlers();
-            value = eventManager.handlers('get');
+            % ID = getCallbackID(@disp)
+            funcID = func2str(callback);
+            funcDetails = functions(callback);
+            if strcmp(funcDetails.type, 'anonymous')
+                workspace = functions(callback).workspace{:};
+                if isfield(workspace, 'obj') &&  isprop(workspace.obj, 'callback') && isprop(workspace.obj, 'figureId')
+                    funcID = sprintf('Function ''%s'' on figure %d', func2str(workspace.obj.callback), workspace.obj.figureId);
+                end
+            end
         end
         
-        function setHandlers(value)
-            % Sets the list of registered callbacks
+        function callback = validateCallback(callback)
+            % Validate the given callback and return handle for the
+            % callback
             %
-            % eventManager.setHandlers(handlerList);
-            eventManager.handlers('set', value);
-        end
-
-        function value=handlers(op, newValue)            
-            % Helper to store the list of registered callbacks as static variable
-            %
-            % handlerList = eventManager.handlers('get');
-            % eventManager.handlers('set', handlerList); 
-            persistent handlers
-            
-            if isempty(handlers) 
-                handlers = {};
+            % callback = validateCallback('disp');
+            if ~isa(callback, 'function_handle') && (~isText(callback) || isempty(char(callback)))
+                throw(exceptions.invalidType('Second value must be function name (text) or handle'));
```

Comment:
"Second value" - there only seems to be one input?

Reply:
No fixed